core/capabilities/launcher.go

```diff
@@ -118,6 +118,74 @@ func NewLauncher(
 	}
 }
 
+// CRE-326: Maintain only necessary Don2Don connections:
+//   - Workflow DONs connect only to other DONs that have at least one remote capability
+//   - Capability DONs connect only to workflow DONs
+//
+// Returns boolean as:
+//   - true: filter out
+//   - false: keep
+func filterDon2Don(logger logger.Logger, isCapabilityDON bool, isWorkflowDON bool, d registrysyncer.DON) bool {
+	// Below logic is based on identification who is who using a workflow acceptance flag:
+	//  d.DON.AcceptsWorkflows == true
+	// We identify few cases from the perspective of the node:
+	if isCapabilityDON && isWorkflowDON {
+		// as both workflow & capability DON let's just connect to anything
+		return false
+	}
+	if !isCapabilityDON && !isWorkflowDON {
+		// as none of workflow & capability DON don't use bandwidth
+		return true
+	}
+	if isWorkflowDON && d.DON.AcceptsWorkflows {
```

Comment:
This one is wrong. Despite `d.DON.AcceptsWorkflows == true`, d.DON can still offer public capabilities.
Can you include a unit test for that scenario please?

Instead, I think you can check` len(d.DON.CapabilityConfigurations) > 0`.